src/TableOfContents/index.tsx

```diff
@@ -0,0 +1,226 @@
+import { Button, Drawer, Icon, IconName } from '@blueprintjs/core';
+import cn from 'classnames';
+import * as React from 'react';
+import { useIsMobile } from '../_hooks/useIsMobile';
+import { ComponentsContext } from '../Provider';
+import { ScrollContainer } from '../ScrollContainer';
+import { IContentsNode } from '../types';
+import { deserializeSrn } from '../utils';
+
+export interface ITableOfContents {
```

Comment:
To get around the Provider issue above, let's add a custom row renderer so Elements can wrap each row in a custom link component:

```suggestion
export interface ITableOfContents {

  rowRenderer?: (item: IContentsNode, DefaultRow: React.FC<{ item: IContentsNode }>): React.Element;
```

In Elements the integration would look something like:

```tsx
export const TableOfContents = ({ items }) => {
  const Components = React.useContext(ComponentsContext);
  const contents = useComputeToc(items);

  return (
    <UIKitTableOfContents
      contents={contents}
      rowRenderer={(node, DefaultRow) => {
        return Components.link({ node, children: <DefaultRow item={node} /> });
      }}
    />
  );
};
```